pkg/lib/editor/editor.go

```diff
@@ -1,393 +1,121 @@
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
+	hashed, _ := bcrypt.GenerateFromPassword([]byte(opts.password), 5)
+	authenticator := auth.NewBasicAuthenticator(opts.username, func(user, realm string) string {
+		if user == opts.username {
+			return string(hashed)
 		}
+		return ""
+	})
 
-		// Get filenames in directory
-		var certs []string
-		err = filepath.Walk(configPath, func(path string, info os.FileInfo, err error) error {
-			if info.IsDir() {
-				return nil
-			}
-			certs = append(certs, path)
-			return nil
-		})
-
-		// Build response
-		resp := map[string]interface{}{
-			"config.yaml": conf,
-			"certs":       certs,
-		}
-		var json = jsoniter.ConfigCompatibleWithStandardLibrary
-		js, err := json.Marshal(resp)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
+	mime.AddExtensionType(".css", "text/css; charset=utf-8")
+	mime.AddExtensionType(".js", "application/javascript; charset=utf-8")
 
-		w.Header().Add("Content-Type", "application/json")
-		w.Write(js)
+	log.Printf("Running the configuration editor on port %v with username %s", opts.port, opts.username)
+	if opts.operatorEndpoint != "" {
+		log.Printf("Using Operator Endpoint: " + opts.operatorEndpoint)
 	}
-}
 
-func certificateHandler(configDir string) func(http.ResponseWriter, *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		switch r.Method {
-		case "PUT":
-			err := r.ParseMultipartForm(10 << 20)
-			if err != nil {
-				fmt.Printf("error parsing request body as form data: %s", err.Error())
-				http.Error(w, err.Error(), http.StatusBadRequest)
-				return
-			}
+	r := chi.NewRouter()
+	r.Use(middleware.Logger)
 
-			file, handler, err := r.FormFile("ca.crt")
-			if err != nil {
-				fmt.Printf("error parsing request body for certificate: %s", err.Error())
-				http.Error(w, err.Error(), http.StatusBadRequest)
-				return
-			}
-			defer file.Close()
-			// FIXME(alecmerdler): Debugging
-			fmt.Printf("Uploaded File: %+v\n", handler.Filename)
-			fmt.Printf("File Size: %+v\n", handler.Size)
-			fmt.Printf("MIME Header: %+v\n", handler.Header)
+	// Function handlers
+	r.Get("/", rootHandler(opts))
+	r.Get("/api/v1/config", auth.JustCheck(authenticator, getMountedConfigBundle(opts)))
+	r.Post("/api/v1/config/validate", auth.JustCheck(authenticator, validateConfigBundle(opts)))
+	r.Post("/api/v1/config/download", auth.JustCheck(authenticator, downloadConfigBundle(opts)))
+	r.Post("/api/v1/config/operator", auth.JustCheck(authenticator, commitToOperator(opts)))
 
-			fileBytes, err := ioutil.ReadAll(file)
-			if err != nil {
-				fmt.Printf("error reading certificate file: %s", err.Error())
-				http.Error(w, err.Error(), http.StatusInternalServerError)
-				return
-			}
+	r.Get("/swagger/*", httpSwagger.Handler(
+		httpSwagger.URL("http://localhost:7070/docs/swagger.json"), // FIXME(jonathan) - This can eventually be changed to the github link to this file.
```

Comment:
This doesn't work when using a non-local hostname. Can you change it to `/docs/swagger.json`?